R/importM.COSMIC.R

```diff
@@ -0,0 +1,348 @@
+    # dataHGNC <- readr::read_delim(file = fNameHGNC,
+    #                               delim = "\t",
+    #                               col_types = readMask,
+    #                               n_max = Inf)
+    #
+    # # Create a map of Cosmic IDs to HGNC IDs
+    # HGNCmap <- dataHGNC[["HGNC_ID"]]
+    # names(HGNCmap) <- dataHGNC[["COSMIC_ID"]]
+
+
+    # ==== READ COSMIC CNV DATA ===============================================
+
+    if (!silent) {
+        cat("Reading CNV data from ", fNameCNV, " ...\n")
+    }
+
+    # Read header line from fNameCNV file
+    tmpHeader <- readLines(fNameCNV, n = 1)
+
+    # Create character vector of col names from header string
+    header <- unlist(strsplit(tmpHeader[1], "\t"))
+
+    # The required columns from Cosmic CNV table
+    required <- c("ID_GENE", "ID_SAMPLE", "ID_TUMOUR", "TOTAL_CN")
+
+    # Find the column numbers in the header that match headings in required
+    iCol <- which(header %in% required)
+
+    # Create a column mask for readr::read_delim()
+    readMask <- paste(paste(rep("_", length(header)),
+                            collapse = ""),
+                      sep = "")
+
+    # Set target columns to characters
+    for (col in iCol) {
+        substr(readMask, col, col) <- "c"
+    }
+
+    # Read only the selected columns from Cosmic CNV file
+    dataCNV <- readr::read_delim(file = fNameCNV,
+                                 delim = "\t",
+                                 col_types = readMask,
+                                 n_max = Inf)
+
+    if (!silent) {
+        cat("Done reading from ", fNameCNV, ".\n")
+    }
+
+    # ==== MAP COSMIC IDs TO HGNC GENE SYMBOLS ================================
+
+    ## Used the following call to generate fastMapCosmic.rds:
+    ## fastMapGenerate("hgnc_complete_set.txt", "symbol",
+    ##                 "cosmic", type = "Cosmic",
+    ##                 outputName = "../inst/extdata/fastMapCosmic.rds")
+
+    # load fastMap hash table
+    fastMapCosmic <- readRDS(system.file("extdata",
+                                       "fastMapCosmic.rds",
+                                       package = "rete"))
+
+    # Validate that the type of the hash table is "Cosmic"
+    if (!is.null(attributes(fastMapCosmic)$type)) {
+        if ("Cosmic" != attributes(fastMapCosmic)$type) {
+            stop(sprintf("Expected hash table type: Cosmic.\nSupplied hash table type: %s",
+                         attributes(fastMapCosmic)$type))
+        }
+    } else {
+        stop("Supplied hash table does not have a type attribute.")
+    }
+
+
+    # ==== CONSTRUCT rCNA OBJECT ==============================================
+
+    if (!silent) {
+        cat("Generating rCNA object using CNA data in ", fNameCNV, ".\n")
+    }
+
+    barcodes <- c()
+    hgncSym <- c()
+
+    for (i in 1:nrow(dataCNV)) {
+
+        # Generate all barcodes for the mutations in dataCNV
+        barcodes <- append(barcodes,
```

Comment:
Barcodes are column headers according to what is written in the design document? Why are you building a vector of **column** headers in a loop that iterates over data **rows**?